src/app/help/help-detail/help-detail.component.ts

```diff
@@ -41,14 +42,18 @@ export class HelpDetailComponent implements OnChanges {
    *  SAVING
    */
   public onSave() {
-    this.changeMode === 'NEW' ? this.createHelpItem() : this.updateHelpItem()
+    const helpObject = { ...this.helpFormComponent.formGroup.value }
+    helpObject.productName = helpObject.product.name
+    delete helpObject.product
+
+    this.changeMode === 'NEW' ? this.createHelpItem(helpObject) : this.updateHelpItem(helpObject)
```

Comment:
Why is that needed?

Reply:
Till now we saved the form exactly as it was (a set of simple fields). After adding combobox, we bind it with an object containing name and displayName. However we still save only name, so before saving we map it